docs/topic-guides/native-api.txt

```diff
@@ -0,0 +1,88 @@
+.. _native-api:
+
+Native API (experimental)
+=========================
+
+.. warning:
+
+    The Native API is currently an experimental API.
+
+The Native API allows developers to seamlessly move between training in local
+Python scripts and training at cluster-scale on a Determined cluster. It also
+provides an interface to train ``tf.keras`` and ``tf.estimator`` models using
+idiomatic framework patterns, reducing (or eliminating) the effort to port
+model code for use with Determined.
+
+In this guide, we'll cover what happens under the hood when a user submits an
+experiment to Determined using the Native API. This topic guide is meant as
+deep-dive for advanced users -- for a tutorial on getting started with the
+Native API, see :ref:`tutorials_native-api`.
+
+Life of a Native API Experiment
+===============================
+
+.. image:: /assets/images/life-of-a-native-experiment.png
+
+.. TODO: Add a link to show what an experiment in local development mode looks like
+
+The diagram above demonstrates the flow of execution when an experiment is
+created in :py:class:`determined.experimental.Mode.CLUSTER
+<determined.experimental.Mode>` mode.
+
+The Python script will first be executed by us in the *User environment*, such
+as a Python virtualenv or a Jupyter notebook where the ``determined`` Python
+package is installed. When the code is executed, ``init()`` will package up all
+the files in the provided context directory, and create an experiment by
+submitting them along with the :ref:`experiment configuration
+<terminology-concepts>` in a network request to the Determined master. Note
+that any code that comes after the ``init()`` call is *not* executed in our
+local environment -- in this case building and compiling the model is only done
+on the Determined cluster.
+
+Once the Determined master has initialized a *Trial Runner environment*, the
+user script is re-executed from the beginning using `runpy.run_path
```

Comment:
"user script is re-executed from the beginning using runpy.run_path" -> "Determined loads the model and other training-related code by using `runpy.run_path() <https://docs.python.org/3/library/runpy.html#runpy.run_path>`_."

Reply:
I prefer being explicit that the user script is re-executed from the beginning. "Determined loads the model and other training-related code" leaves it more ambiguous imo.

Reply:
got it. My point is the user script is executed twice but for different purposes. But I can see your point